concepts/type-theory-glossary.md

```diff
@@ -0,0 +1,396 @@
+A quick Primer and Glossary of Type Theory
+==========================================
+
+This document provides a quick overview of "type theory", and specifically,
+how we use it to reason about data and design in IPLD.
+
+We take "type theory" to mean primarily: a way of reasoning about programs
+and data by counting how many possible states they may have.
+We call this counting of possible, inhabitable states counting "cardinality".
```

Comment:
```suggestion
We call the number of possible, inhabitable states "cardinality".
```

Reply:
You're right that the phrasing is stilted, and perhaps this could still be improved... but I think I'm going to leave this one for now, because I really want to hammer the association with the concept and action of _counting_ rather than the existence of _a_ number.